include/neso_particles/compute_target.hpp

```diff
@@ -502,7 +503,7 @@ template <typename T> class BufferShared : public BufferBase<T> {
     return static_cast<T *>(
         sycl::malloc_shared(num_bytes, this->sycl_target->queue));
   }
-  virtual inline void free_wrapper(T *ptr) override {
+  virtual inline void free_wrapper([[maybe_unused]] T *ptr) override {
```

Comment:
I think this one is an actual bug and the free line should be freeing `ptr` not `this->ptr`?